DependencyInjection/Configuration.php

```diff
@@ -208,4 +210,56 @@ private function addVisitorsSection(NodeBuilder $builder)
             ->end()
         ;
     }
+
+    private function addContextSection(NodeBuilder $builder)
+    {
+        $builder
+            ->arrayNode('context')
+                ->addDefaultsIfNotSet()
+                ->children()
```

Comment:
what about having also just `id`, to specify just a service id as context factory? and when specified, the rest of options are just ignored 
It will be super flexible. 

I understand this is an extra requirement that can be implemented later, so this is just a suggestion...